In [1]:
import pandas as pd

## Predicting Concrete strength

In [2]:
concrete = pd.read_csv('slump_test.data',sep=',')
concrete.info()
concrete.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 103 entries, 0 to 102
Data columns (total 11 columns):
 #   Column                              Non-Null Count  Dtype  
---  ------                              --------------  -----  
 0   No                                  103 non-null    int64  
 1   Cement                              103 non-null    float64
 2   Slag                                103 non-null    float64
 3   Fly ash                             103 non-null    float64
 4   Water                               103 non-null    float64
 5   SP                                  103 non-null    float64
 6   Coarse Aggr.                        103 non-null    float64
 7   Fine Aggr.                          103 non-null    float64
 8   SLUMP(cm)                           103 non-null    float64
 9   FLOW(cm)                            103 non-null    float64
 10  Compressive Strength (28-day)(Mpa)  103 non-null    float64
dtypes: float64(10), int64(1)
memory usage: 9.0 KB

No  Cement   Slag  Fly ash  Water    SP  Coarse Aggr.  Fine Aggr.  \
0   1   273.0   82.0    105.0  210.0   9.0         904.0       680.0   
1   2   163.0  149.0    191.0  180.0  12.0         843.0       746.0   
2   3   162.0  148.0    191.0  179.0  16.0         840.0       743.0   
3   4   162.0  148.0    190.0  179.0  19.0         838.0       741.0   
4   5   154.0  112.0    144.0  220.0  10.0         923.0       658.0   

   SLUMP(cm)  FLOW(cm)  Compressive Strength (28-day)(Mpa)  
0       23.0      62.0                               34.99  
1        0.0      20.0                               41.14  
2        1.0      20.0                               41.81  
3        3.0      21.5                               42.08  
4       20.0      64.0                               26.82

The dataset information are the following : 
The data set includes 103 data points. There are 7 input variables, and 3 output 
variables in the data set.  The initial data set included 78 data. After several 
years, we got 25 new data points.

----------------------------------------------------------------------------------------

Attribute Information:

Input variables (7)(component kg in one M^3 concrete):
Cement
Slag
Fly ash
Water
SP
Coarse Aggr.
Fine Aggr.

Output variables (3):
SLUMP (cm)
FLOW (cm)
28-day Compressive Strength (Mpa) 



The aim of the project is to predict the variables as a function of the concrete composition. In this aim, carry out the following workflow.

In [1]:
import pandas as pd
from matplotlib import pyplot as plt

1. Split the dataset in X and Y and in training and test set.

2. Define the model (use tensorflow)

3. Train the model

4. Look at the training 

5. Compare with experiment

And now play with it to improve the procedure : standardize the input data, change the model ... 